channeldb/channel.go

```diff
@@ -3558,6 +3625,117 @@ func DeserializeChainActions(r io.Reader) (map[string][]HTLC,
 	return chainAction, nil
 }
 
+// serializeLocalForceCloseInsights writes out the passed
+// localForceCloseInsights to the writer.
+func serializeLocalForceCloseInsights(w io.Writer,
+	lc *LocalForceCloseInsights) error {
+
+	localForceCloseReasonByte := []byte(lc.Initiator)
+
+	tlvStream, err := tlv.NewStream(
+		tlv.MakePrimitiveRecord(localForceCloseInitiatorType,
+			&localForceCloseReasonByte),
+		tlv.MakeDynamicRecord(
+			chainActionType, &lc.ChainActions,
+			HtlcActionRecordSize(&lc.ChainActions),
+			HtlcActionEncoder,
+			HtlcActionDecoder,
+		),
+		tlv.MakePrimitiveRecord(broadCastHeightType,
+			&lc.BroadcastHeight),
+	)
+
+	if err != nil {
+		return err
+	}
+	var b bytes.Buffer
+	err = tlvStream.Encode(&b)
+	if err != nil {
+		return err
+	}
+
+	err = binary.Write(w, byteOrder, uint64(b.Len()))
+	if err != nil {
+		return err
+	}
+
+	if _, err = w.Write(b.Bytes()); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// deSerializeLocalForceCloseInsights reads out the LocalForceCloseInsights from
+// the passed reader.
+func deSerializeLocalForceCloseInsights(r io.Reader) (*LocalForceCloseInsights,
```

Comment:
Thanks

Reply:
https://github.com/lightningnetwork/lnd/pull/8072#discussion_r1430130929
Are we going to store the remote force close here though, maybe the force close initiator would just state `remote force close` in that case. Looking at it again maybe it can do with a name change.